### PR TITLE
Added autoOpen items on hover 

### PR DESCRIPTION
**Pull request by siddhant1**
I have verified that this pull request:

* [x ] has no linting errors (`npm run lint`)
* [ x] is from a uniquely-named feature branch and has been rebased on top of the latest master. (If I was asked to make more changes, I have made sure to rebase onto master then too)
* [x ] is descriptively named and links to an issue number, i.e. `Fixes #123`

Added as discussed in https://github.com/processing/p5.js-web-editor/issues/871#issuecomment-472167759

### CODE REVIEW

**Comment by siddhant1** [comment]
Also added some checks because they were throwing error in the console

**Comment by siddhant1** [comment]
ezgif-1-23499fb0035e
Screencast for referance

**Comment by siddhant1** [comment]
In screencast its not clear but I closed the tab to show its working with tab closed too

**Comment by catarak** [comment]
this is working great and i like this ux! could you remove the heading in the dropdown, as it feels like duplicate information?

**Comment by siddhant1** [comment]
Pushed 
image

File: client/styles/components/_nav.scss [hunk 1]
```diff
@@ -76,7 +76,7 @@
 		display: flex;
 		position: absolute;
 		flex-direction: column;
-  	top: 4px;
+  	top: 40px;
```
Comment (plxity) [inline client/styles/components/_nav.scss]: Indent it properly

File: client/components/Nav.jsx [hunk 2]
```diff
@@ -239,18 +239,18 @@ class Nav extends React.PureComponent {
...
-              <button
-                onClick={this.toggleDropdown.bind(this, 'edit')}
-                className="nav__dropdown-heading"
-              >
-                <span>Edit</span>
-                <InlineSVG src={triangleUrl} />
-              </button>
+            <ul
+              className="nav__dropdown"
+            >
```
Comment (plxity) [inline client/components/Nav.jsx]: Move everything in one line.

File: client/styles/components/_nav.scss (hunk 1 above)
Comment (siddhant1) [reply client/styles/components/_nav.scss]: Not Sure , its showing perfectly in my code editor , maybe github diff loader is showing it this way?

Comment (siddhant1) [reply client/styles/components/_nav.scss]: > Indent it properly
Not Sure , its showing perfectly in my code editor , maybe github diff loader is showing it this way?

Comment (catarak) [reply client/styles/components/_nav.scss]: also this should be a rem measurement, not a pixel measurement.

**Comment by catarak** [comment]
this is looking good. a couple of things:
1. when you hover over the first nav item, the lack of border-radius is over a border radius and it looks weird: 
Screen Shot 2019-03-20 at 1 08 10 PM
2. i think that the upper left hand corner should have no border radius. it looks weird, and i took a look at how Google docs handles it, and that's what they do. 
Screen Shot 2019-03-20 at 1 09 33 PM

File: client/styles/components/_nav.scss (hunk 1 above)
Comment (siddhant1) [reply client/styles/components/_nav.scss]: This was already a pixel measurement , should I change it?

**Comment by siddhant1** [comment]
I didn't get your point 1 , also should I remove top-left border radius from the drop-down?

File: client/styles/components/_nav.scss (hunk 1 above)
Comment (catarak) [reply client/styles/components/_nav.scss]: yes!

**Comment by catarak** [comment]
re: point 1, you have to look really closely:
Screen Shot 2019-03-21 at 6 00 37 PM
if you look at the upper right hand corner, the pink overlay doesn't have a border radius, whereas its container does. you need to give the pink overlay a border radius.

**Comment by siddhant1** [comment]
image
There is a very small (barely noticeable )  lack of border-radius

**Comment by siddhant1** [comment]
image
image

Fixed it !

File: client/components/Nav.jsx (hunk 2 above)
Comment (catarak) [reply client/components/Nav.jsx]: i agree—this doesn't need to be on three lines.

**Comment by catarak** [comment]
another UX thing—when a user hits the ESC key, could it close the nav dropdown if it's open? as i was playing around with this PR i instinctively did that, and then i checked with google docs to see if they follow that UX, and it turns out, they do.

**Comment by siddhant1** [comment]
Sure , I think adding an event listener od esc will do the work , Also I was thinking that all the dropdowns should close when the loose focus or on `esc`

**Comment by catarak** [comment]
i also noticed that the "my account" dropdown still has a header:
Screen Shot 2019-03-26 at 12 32 15 PM
can you remove it?

**Comment by siddhant1** [comment]
Whoops ! Sure I will remove it

**Comment by siddhant1** [comment]
>https://github.com/processing/p5.js-web-editor/pull/900#issuecomment-476731101

I think I should make another pr with this feature

**Comment by catarak** [comment]
> > [#900 (comment)](https://github.com/processing/p5.js-web-editor/pull/900#issuecomment-476731101)
> 
> I think I should make another pr with this feature

okay! that makes sense to me.

**Comment by catarak** [comment]
this is working great for me! i'm gonna merge this.

**Pull request merged**